src/lib.rs

```diff
@@ -51,6 +51,95 @@ lazy_static! {
     static ref PATCHED_IMAGE_DOCKER_TAG: OsString = OsString::from_str("mx-tester/synapse").unwrap();
 }
 
+#[derive(Debug, Deserialize)]
+pub struct DockerConfig {
+    #[serde(default)]
+    /// A docker network to run the synape containers on.
+    /// If the network does not exist it will be created.
+    /// If a network is provided, the synapse container will be given the hostname `synapse`.
+    /// Defaults to None.
+    pub docker_network: Option<String>,
+
+    /// The hostname to give the synapse container on the docker network, if the docker network has been provided.
+    /// Defaults to `synapse` but will not be used unless a network is provided in docker_network.
+    pub hostname: String,
+
+    /// The docker port mapping configuration to use for the synapse container e.g. `9999:9999`.
+    /// Defaults to 9999:9999
+    pub port_mapping: String,
+}
+
+impl Default for DockerConfig {
+    fn default() -> DockerConfig {
+        DockerConfig {
+            docker_network: None,
```

Comment:
Why `None` and not `"synapse"`?

Reply:
This is the name of a docker network that the user of mx-tester can provide IF they have set one up, or they want one to be setup with that name. It's an option so that None can be used so that a network isn't specified when mx-tester uses docker run so that docker just uses its default.

Reply:
Ah, my bad, I had misunderstood the documentation above.